### ISSUE TITLE
Reload skills

### ISSUE THREAD

**Issue by colonelkrud** [opsdroid/opsdroid#655]
Perhaps as part of #643  we should rebuild the reload function. 

When developing, sometimes I want to reload a skill many times to text each part of functionality. Currently the only way to reload skills is to restart the bot. My suggestion is a new method of loading skills that allows for a single skill or all skills to be reloaded while the bot is live. It could even recheck the config file and redownload non local skills when no single skill is named - meaning skills could be added or removed while the bot is running. This should be open to the skills themselves as a function of the core such that it can be called from within a skill. 

An example would be a dev skill which listens for various commands in a dev room, checks the user is on an approved user list, then runs the reload <config|(skill name)>. This would allow for rapid development and testing of skills.

**Comment by FabioRosado** [comment]
I believe the issue with the reload and why we decided to remove it was because it caused some issues with the loader.

**Comment by colonelkrud** [comment]
True.. but if we are looking at improving the loader, this should be something considered from the beginning.

**Comment by jacobtomlinson** [comment]
The reload was removed because it was poorly implemented (by me) and caused more problems than it solved, so the decision was to take it back out again. However I would like to see this functionality in opsdroid, just done properly 😂.

One package to look at for inspiration would be [appdaemon](https://github.com/home-assistant/appdaemon). They have auto loading and 'auto reload on file changed' for their plugins.
